### PR TITLE
fix: ownership-aware Q&A/RFI filters + finite flow-layout measurement (#1200, #1203)

### LINKED ISSUE #1200: [Chat][Bug] Q&A review and RFI filters show unrelated open threads
Status: OPEN

## Problem
The native Chat Q&A/RFI list filters are over-broad and can send users to the wrong work queue:

1. `IOSQuestionsPage` labels a smart-card filter as `Needs My Review`, but the count and filter both include every `open` Q&A thread instead of checking whether the current user/role is actually the responder.
   - Count: `Weird Parts IOS/Weird Parts IOS/Features/Chat/IOSQuestionsPage.swift:103`
   - Filter: `Weird Parts IOS/Weird Parts IOS/Features/Chat/IOSQuestionsPage.swift:199`

2. `IOSRFIListPage` is documented as office-only RFIs where “RFIs are Q&A threads that have been escalated to the office level,” but `loadData()` loads `service.listQAThreads()` with no status/level filter, then `Open` includes every open/escalated Q&A thread.
   - Documentation/intent: `Weird Parts IOS/Weird Parts IOS/Features/Chat/IOSRFIListPage.swift:4-8`
   - Count/filter: `Weird Parts IOS/Weird Parts IOS/Features/Chat/IOSRFIListPage.swift:102-108`, `228-236`
   - Load source: `Weird Parts IOS/Weird Parts IOS/Features/Chat/IOSRFIListPage.swift:331-334`

This was found during the WEI-4132 Bug Hunt + GitHub Issue Fisher pass using `docs/plans/usability-hunter-plan.md` guidance for navigation/feedback bugs.

## Impact
Field/office users can see unrelated Q&A items in “Needs My Review” or RFI queues, making the triage counts unreliable and increasing the chance urgent questions assigned to the user are buried among unrelated open threads.

## Steps to reproduce
1. Create or seed multiple Q&A threads at different escalation levels / asked-by users.
2. Open Chat → Q&A.
3. Tap `Needs My Review`.
4. Observe that every `status == "open"` thread is shown, regardless of whether it needs the current user's review.
5. Open Chat → RFIs.
6. Observe that all Q&A threads from `listQAThreads()` are loaded and the Open card includes every open/escalated thread, not just office-level RFIs.

## Expected behavior
- `Needs My Review` should only count/show threads actionable by the current user or their role/escalation level.
- RFI should only count/show Q&A threads that are actually RFIs/office-level escalations plus supplier questions.
- Empty states should distinguish “no work assigned to you” from “no questions exist.”

## Suggested fix
Add service-level query helpers or row metadata for actionable Q&A/RFI ownership, then update the smart-card counts and filtered arrays to use those helpers instead of client-side `status == "open"` checks. Add tests for mixed Q&A data with field, lead, manager, and office-level threads.

## Acceptance criteria
- [ ] `Needs My Review` only includes threads requiring the current user/role to respond.
- [ ] RFI list only includes office-level/RFI Q&A threads and supplier RFI questions.
- [ ] Smart-card counts match the visible filtered rows.
- [ ] Regression tests cover mixed status/level/assignee Q&A data.

Related planning context: #79

### LINKED ISSUE #1203: [Layout][Bug] Custom flow layouts can report infinite measured width
Status: OPEN

## Problem
Several custom SwiftUI `Layout` implementations use `proposal.width ?? .infinity` during `sizeThatFits`, then return that same width in their measured `CGSize`. When SwiftUI asks for an unconstrained measurement, these layouts can report `CGSize(width: .infinity, ...)` instead of their finite content width.

## Evidence
- `Weird Parts IOS/Weird Parts IOS/Features/Warehouse/IOSMovementWizard.swift:1185` sets `maxWidth = proposal.width ?? .infinity` and `:1203` returns `CGSize(width: maxWidth, ...)`.
- `Weird Parts IOS/Weird Parts IOS/Features/Parts/TypeBrandColorSection.swift:706` passes `proposal.width ?? .infinity`; `:743` returns `CGSize(width: width, ...)`.
- `Weird Parts IOS/Weird Parts IOS/Features/Scheduling/IOSScheduleConfigPage.swift:564` passes `proposal.width ?? .infinity`; `:601` returns `CGSize(width: width, ...)`.
- `Weird Parts IOS/Weird Parts IOS/Features/Parts/CompanionSandboxSheet.swift:409` uses the same unbounded width for wrapping decisions; it returns a finite `maxWidth`, but size measurement still disables wrapping when the proposal is nil, so it should be audited with the same helper.

## Impact
Affected chip/flow rows can mis-measure in sheets, forms, scroll views, or other unconstrained containers. On iPhone/iPad this can cause horizontal overflow, clipped controls, broken wrapping, or SwiftUI layout warnings/crashes when the layout engine receives a non-finite size.

## Expected behavior
Custom flow layouts should always return finite measured sizes and wrap against a concrete width. If the proposed width is nil or non-finite, use a finite fallback based on content width or the available bounds, and share one safe wrapping helper to avoid drift.

## Acceptance criteria
- [ ] Flow layouts never return `.infinity` or non-finite width/height from `sizeThatFits`.
- [ ] Movement reason chips, brand/type/color chips, schedule config chips, and companion sandbox chips wrap correctly at iPhone, iPad, and desktop-width preview sizes.
- [ ] Add a regression test or preview-driven assertion covering nil/unconstrained proposal behavior for the shared flow-layout helper.
- [ ] Confirm no horizontal overflow in the affected screens.

Found during WEI-4132 BugHunter usability sweep.

### PR DESCRIPTION
**Pull request by xXKillerNoobYT**
Closes #1200
Closes #1203

Plan: docs/plans/beta-readiness-issue-resolution-2026-07.md §3 Batch N (+ #1203 folded into this batch).

## What changed

**#1200 — Q&A review and RFI filters show unrelated open threads**
- `ChatService.actionableQALevels(userId:)` (core): maps a user's active hats to the escalation levels they respond at — Worker→worker, Lead→lead, Manager→manager, Office→office, Admin→entire chain; Apprentice/Grunt/custom hats are askers, not responders. Missing hat tables degrade to empty (existing `isTableNotFoundError` pattern).
- `ChatService.listQAThreadsNeedingReview(userId:)` (core): open/escalated threads at one of the user's levels, excluding the user's own questions **unless** the thread was pushed back down to them (detected via `qa_escalations` from-level > to-level records), so push-back feedback still lands in the asker's queue.
- `ChatService.listQAThreads(status:levels:)` (core): optional escalation-level filter; `nil` preserves the old all-levels behavior, so existing call sites are untouched.
- `IOSQuestionsPage`: the "Needs My Review" smart-card count and filtered rows both use the new ownership-aware ID set (previously every `status == "open"` thread), so the count always matches the visible rows. Empty states now distinguish "No Questions" / "Nothing Needs Your Review" / "No Matching Questions".
- `IOSRFIListPage`: `loadData()` now loads `listQAThreads(levels: ["office"])` — only office-level escalations (true RFIs) plus supplier questions, matching the page's documented intent. Empty states distinguish "No RFIs" from "No Matching RFIs".

**#1203 — Custom flow layouts can report infinite measured width**
- `FlowLayoutMath` (core): shared pure-geometry wrap helper. `resolveWrapWidth` guards nil/`.infinity`/NaN/non-positive proposals; unconstrained proposals lay out a single row and report its **finite** content width instead of echoing `.infinity` back. Constrained behavior (wrap + report the bound) is unchanged from the previous implementations.
- `FlowChipLayout` (app `Shared/`): single SwiftUI `Layout` adapter over `FlowLayoutMath`, replacing the four duplicated implementations flagged in the issue: `MovementWizardFlowLayout` (IOSMovementWizard), `BrandFlowLayout` (TypeBrandColorSection — now a typealias; the struct had no call sites), `FlowLayout` (IOSScheduleConfigPage), `WrappedHStack` (CompanionSandboxSheet). One code path = no future drift.
- Regression tests cover nil/infinite/NaN/zero/negative proposals, wrapping, tallest-row height, oversized items, and empty input.

## Test evidence
- `cd core && swift build` → `Build complete! (23.50s)`
- `cd core && swift test --filter "FlowLayoutMathTests|QAReviewFilterTests"` → `Test run with 17 tests in 2 suites passed` (9 QA review-filter tests incl. hat mapping, per-level queues, asker exclusion, push-back return, resolved exclusion, RFI level filter; 8 flow-layout math tests)
- `cd core && swift test --filter "ChatService"` → `Test run with 61 tests in 1 suite passed` (no regressions in the existing suite)
- `xcrun swiftc -parse` on all 7 touched app-target files → clean (app target uses filesystem-synchronized groups, so the new `Shared/FlowChipLayout.swift` is picked up automatically)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Review follow-up (commit 5d11b333e)

**[blocker] RFI creation dead-ended against the office-level list.** The RFI page's '+' form created threads hard-coded at `'worker'` level, so after this PR's office-level filter the `onSubmitted` reload could never show the just-created thread — the user landed on the "No questions have been escalated to the office level" empty state.

- `ChatService.createQAThread` gains a validated `level:` parameter (default `"worker"`, so every existing call site is unchanged). Unknown levels throw the new `ChatError.invalidEscalationLevel` before any insert.
- `IOSQAQuestionForm` passes a new `escalationLevel` property through and titles itself "New RFI" when creating at office level; `dismiss()` now runs before `onSubmitted()` per the sheet convention.
- `IOSRFIListPage` presents the form with `escalationLevel: "office"` — the reload immediately shows the created RFI.
- +3 regression tests in `QAReviewFilterTests` (office-level creation lands in the RFI queue and not the worker queue; default stays `worker`; invalid level throws without inserting).

Follow-up test evidence:
- `cd core && swift build` → `Build complete! (30.56s)`
- `cd core && swift test --filter "QAReviewFilterTests"` → `Test run with 12 tests in 1 suite passed`
- `cd core && swift test --filter "ChatService"` → `Test run with 61 tests in 1 suite passed`
- `xcrun swiftc -parse` on both touched app files → clean

### CODE REVIEW

File: core/Sources/WiredPartCore/Services/ChatService.swift [hunk 1]
```diff
@@ -494,25 +507,149 @@ public final class ChatService: Sendable {
...
+                return rows.map { row in
+                    QAThreadRow(
+                        id: row["id"] ?? 0,
+                        jobId: row["job_id"] ?? 0,
+                        askedById: row["asked_by"] as Int64?,
+                        question: row["subject"] ?? "",
+                        askedByName: row["asked_by_name"] ?? "Unknown",
+                        currentLevel: row["current_level"] ?? "field",
+                        status: row["status"] ?? "open",
+                        priority: row["priority"] ?? "normal",
```
[the remark this answers is not in the document]
Comment (xXKillerNoobYT) [reply core/Sources/WiredPartCore/Services/ChatService.swift:610]: Fixed in 1a1013299: changed the `?? "field"` fallback to `?? "worker"` in `listQAThreadsNeedingReview`, matching the `qa_threads.current_level` schema default and the bottom of `qaEscalationLevels` (which does not contain "field" at all). Also aligned the same fallback in the two pre-existing sibling query functions (`listQAThreads`) in the same file for consistency.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
